azure-iot-provisioning-devicessdk/azure/iot/provisioning/devicessdk/device/sk_registration_client.py

```diff
@@ -0,0 +1,35 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""
+This module contains one of the implementations of the Registration Client which uses Symmetric Key authentication.
+"""
+import logging
+from .registration_client import RegistrationClient
+
+logger = logging.getLogger(__name__)
+
+
+class SymmetricKeyRegistrationClient(RegistrationClient):
+    """
+    Client which can be used to run the registration of a device using Symmetric Key authentication.
+    """
+
+    def __init__(self, transport):
+        """
+        Initializer for the Symmetric Key Registration Client
+        """
+        RegistrationClient.__init__(self, transport)
```

Comment:
Use super

Reply:
I thought the 2 super formats for python 2 and python 3 are not same and this is the only one that works for both. I can find out more when unit tests are added.

Python 2 format for super is
`super().__init__(arg1, arg2)`

Python 3 super is
`super(ChildClass, self).__init__(arg1, arg2)`

Reply:
That's backwards, it's Python 2 that is super(ChildClass, self).__init__(arg1, arg2). And that version is also compatible in 3. 

---
In reply to: [267876918](https://github.com/Azure/azure-iot-sdk-python-preview/pull/59#discussion_r267876918) [](ancestors = 267876918)